v3iofs/fs.py

```diff
@@ -52,29 +58,79 @@ def __init__(self, v3io_api=None, v3io_access_key=None, **kw):
             transport_kind='requests',
         )
 
+    def _details(self, contents, **kwargs):
+        pathlist = []
+        for c in contents:
+            data = {}
+            data['name'] = c['name'].lstrip("/")
+            if c['size'] is not None:
+                data['type'] = 'file'
+                data['size'] = c['size']
+            else:
+                data['type'] = 'directory'
+                data['size']= 0
+            pathlist.append(data)
+        return pathlist
+
     def ls(self, path, detail=True, **kwargs):
         """Lists files & directories under path"""
+        
         container, path = split_container(path)
-
-        if not container:
+        if container == "":
             return self._list_containers(detail)
-
-        resp = self._client.get_container_contents(
-            container=container,
-            path=path,
-            get_all_attributes=True,
-            raise_for_status=[HTTPStatus.OK],
-        )
-
-        prefixes = resp.output.common_prefixes  # directories
-        fn = prefix_info if detail else prefix_path
-        dirs = [fn(container, p) for p in prefixes]
-
-        objects = resp.output.contents  # files
-        fn = object_info if detail else object_path
-        files = [fn(container, o) for o in objects]
-
-        return dirs + files
+        elif container and path == "":
+            containers = self._list_containers(detail=True)
+            containers = [c['name'] for c in containers]
+            if not container in containers:
+                raise FileNotFoundError("Container not found!!")
+        try:
+            resp = self._client.get_container_contents(
+                container=container,
+                path=path,
+                get_all_attributes=True,
+                raise_for_status=[HTTPStatus.OK],
+            )
+
+            # Try to fetch a list of directories, else return empty
+            try:
+                prefixes = resp.output.common_prefixes
+                dirs = [prefix_info(container, p) for p in prefixes]
+            except:
+                dirs = []
+
+            # Try to fetch a list of files, else return empty
+            try:
+                objects = resp.output.contents
+                files = [object_info(container, o) for o in objects]
+            except:
+                files = []
+            pathlist = dirs + files
+            if not pathlist:
+                raise FileNotFoundError(f"Nothing found in {path}")
+        except:
+            logger.debug("Nothing found in container.  Moving to top level container")
```

Comment:
I'm not sure this is the logic we want. Also too much code inside an `except` clause.